core/milvus_storage.go

```diff
@@ -1,11 +1,268 @@
 package core
 
+import (
+	"bytes"
+	"context"
+	"errors"
+	"go.uber.org/zap"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/zilliztech/milvus-backup/internal/log"
+	"github.com/zilliztech/milvus-backup/internal/util/errorutil"
+)
+
+type FileReader interface {
+	io.Reader
+	io.Closer
+}
+
 // MilvusStorage is interface to operate milvus data
 type MilvusStorage interface {
+	// Path returns path of @filePath.
+	Path(filePath string) (string, error)
+	// Size returns path of @filePath.
+	Size(filePath string) (int64, error)
+	// Write writes @content to @filePath.
+	Write(filePath string, content []byte) error
+	// MultiWrite writes multi @content to @filePath.
+	MultiWrite(contents map[string][]byte) error
+	// Exist returns true if @filePath exists.
+	Exist(filePath string) (bool, error)
+	// Read reads @filePath and returns content.
+	Read(filePath string) ([]byte, error)
+	// Reader return a reader for @filePath
+	Reader(filePath string) (FileReader, error)
+	// MultiRead reads @filePath and returns content.
+	MultiRead(filePaths []string) ([][]byte, error)
+	ListWithPrefix(prefix string, recursive bool) ([]string, []time.Time, error)
+	// ReadWithPrefix reads files with same @prefix and returns contents.
+	ReadWithPrefix(prefix string) ([]string, [][]byte, error)
+	// ReadAt reads @filePath by offset @off, content stored in @p, return @n as the number of bytes read.
+	// if all bytes are read, @err is io.EOF.
+	// return other error if read failed.
+	ReadAt(filePath string, off int64, length int64) (p []byte, err error)
+	// Remove delete @filePath.
+	Remove(filePath string) error
+	// MultiRemove delete @filePaths.
+	MultiRemove(filePaths []string) error
+	// RemoveWithPrefix remove files with same @prefix.
+	RemoveWithPrefix(prefix string) error
 }
 
 // makes sure MinioMilvusMeta implements `MilvusStorage`
 var _ MilvusStorage = (*MinioMilvusMeta)(nil)
 
 type MinioMilvusMeta struct {
```

Comment:
I think the name is not proper. We operate data here not meta data. How about MinioMilvusStorage?